### ISSUE TITLE
?Enhancement: connect presentations to the participant's ticket

### ISSUE THREAD

**Issue by IGORnvk** [HZ-HBO-ICT/it-conference#605]
Currently the ticket system works okay, but, as was discovered, there is an issue in case the attending student is an as*hole, he can have his ticket scanned at the entrance and just leave the building. So, I thought about a possible solution: what if we can kind of 'connect' the ticket of the user to the presentations he signed up for and make him scan the ticket on the entrance to every presentation. Then, this will resolve two issues:
1. Will be easier for the crew to let in participants to the lecture/workshop, since the system will automatically display if they are eligible for it.
2. Will prevent the issue with leaving the building right after scanning at the entrance, since the admin would be able to see the status of the ticket (how many presentations did the student attend) and determine if the student actually participated.

What do you think @v-stamenova @TimKardol ?

**Comment by v-stamenova** [comment]
I think it sounds pretty reasonable. If I am honest I am down for a challenge to make the next edition a bit more 'data driven' if it makes sense? I think tracking our participant activity might be a good step towards it. @TimKardol what do you say?

**Comment by TimKardol** [comment]
I agree with both of you. This will make the whole process a bit easier.

**Comment by IGORnvk** [comment]
How I see this:
1. On the scanning page (`/moderator/tickets`) add a dropdown before activating a scanner which will ask the location from the user. The dropdown will have the list of rooms and some general location(s), like "entrance", because the ticket should also be scanned when participants enter the building.
2. When the location from the dropdown is chosen (and stored in a Livewire component), a scanner is displayed as usual and tickets can be scanned.
3. When the ticket is scanned, the algorithm will:
    a) Match the time when it was scanned with the closest presentation scheduled in the chosen location (if location is not one of the rooms, proceed with the usual ticket scan).
    b) Check if participant is enrolled for that presentation.
    c) To prevent any types of "cheating" or accidents (like scanning the ticket very early before / late after the beginning of the presentation), some time boundaries could be set up. For instance, allow "checking in" for the presentation only 10-15 minutes before it begins, and deny anything after 1/2 or 1/3 of the duration of the presentation.
4. Additionally, might be useful to make moderators manually change the status of the ticket in case of any problems or malfunctions (let's gooo, making the attendance system).

This way, a person responsible for the upcoming presentation in any room can set up a laptop/tablet or whatever that can scan the tickets at the entrance to the room and if anything happens, alter the system manually.
What do you guys think @TimKardol @v-stamenova ?
